packml_sm/include/packml_sm/abstract_state_machine.h

```diff
@@ -174,121 +174,143 @@ class AbstractStateMachine
   /**
    * @brief Accessor for the duration spent in idle time.
    *
+   * @param is_transaction flag to either calculate for a transaction or a running sum
    * @return double Returns the time spent in the idle state.
    */
-  double getIdleTime();
+  double getIdleTime(bool is_transaction);
```

Comment:
Assigning a default argument value to this signature (and similar below) would allow the function to continue to work as it had previously without forcing an API change.

Reply:
I added default false value to all these signatures, and removed the value when non-incremental stats client code calls them.